sdk/python/feast/feature_store.py

```diff
@@ -254,6 +254,23 @@ def apply(
             partial=True,
         )
 
+    @log_exceptions_and_usage
+    def teardown(self):
+        feature_defs: List[Union[FeatureView, FeatureTable]] = []
+        feature_views = self.list_feature_views()
+        feature_tables = self._registry.list_feature_tables(self.project)
+
+        feature_defs.extend(feature_views)
+        feature_defs.extend(feature_tables)
+
+        entities = self.list_entities()
+
+        self._get_provider().teardown_infra(self.project, feature_defs, entities)
+        for feature_view in feature_views:
```

Comment:
Why are we deleting feature_views and feature_tables but not deleting entities?

Reply:
There's no support to delete the entities in the registry yet: #1694 . But it should exist, and be called from here.

Yes, these lines are intended to tear down the registry, but this behaviour is different from what the CLI does - you can check `teardown` in `repo_operations`, and it doesn't tear down the registry. That should be rationalized (i.e. `repo_operations` should call the feature store method, and the registry *should* be torn down.)